symExec.py

```diff
@@ -860,34 +862,25 @@ def sym_exec_ins(start, instr, stack, mem, global_state, path_conditions_and_var
             first = stack.pop(0)
             second = stack.pop(0)
             third = stack.pop(0)
-            if isinstance(third, (int, long)):
+
+            if contains_only_concrete_values([first, second, third]):
                 if third == 0:
                     computed = 0
                 else:
-                    if not (isinstance(first, (int, long)) and isinstance(second, (int, long))):
-                        # there is one guy that is a symbolic expression
-                        third = BitVecVal(third, 256)
-                        if isinstance(first, (int, long)):
-                            first = BitVecVal(first, 256)
-                        if isinstance(second, (int, long)):
-                            second = BitVecVal(second, 256)
-                    first = to_unsigned(first)
-                    second = to_unsigned(second)
-                    third = to_unsigned(third)
                     computed = (first + second) % third
             else:
-                solver.push()
-                solver.add(Not(third == 0))
-                if solver.check() == unsat:
-                    # it is provable that second is indeed equal to zero
+                first = to_symbolic(first)
+                second = to_symbolic(second)
+                solver.add(third == 0)
+                if solver.check() == sat:
                     computed = 0
                 else:
-                    if isinstance(first, (int, long)):
-                        first = BitVecVal(first, 256)
-                    if isinstance(second, (int, long)):
-                        second = BitVecVal(second, 256)
+                    first = ZeroExt(256, first)
+                    second = ZeroExt(256, second)
+                    third = ZeroExt(256, third)
                     computed = (first + second) % third
-                solver.pop()
+                    computed = Extract(255, 0, computed)
+                solver.reset()
```

Comment:
I dont think using reset is a good idea. For example, using reset would remove the backtracking point created by https://github.com/melonproject/oyente/blob/master/symExec.py#L585